File: backend/models/loan.go
package models

import (
	"time"
)

// Loan 贷款模型
type Loan struct {
	ID                 uint          `gorm:"primaryKey" json:"id"`
	BorrowerName       string        `gorm:"size:100" json:"borrower_name"`
	BankName           string        `gorm:"size:100" json:"bank_name"`
	TotalAmount        float64       `gorm:"column:total_amount" json:"total_amount"`   // 贷款总额（原始金额）
	RemainingAmount    float64       `gorm:"column:remaining_amount" json:"remaining_amount"` // 当前余额
	InterestRate       float64       `json:"interest_rate"`
	StartDate          time.Time     `json:"start_date"`                                // 贷款开始日期
	EndDate            time.Time     `json:"end_date"`                                  // 贷款结束日期
	FirstPaymentDate   *time.Time    `json:"first_payment_date,omitempty"`              // 第一期还款日期（可选）
	PaymentDate        int           `gorm:"default:1" json:"payment_date"`             // 每月还款日（1-31）
	Status             LoanStatus    `gorm:"size:20;default:active" json:"status"`
	PaymentMethod      PaymentMethod `gorm:"size:50;default:等额本息" json:"payment_method"`
	MonthlyPayment     float64       `json:"monthly_payment"`
	CreatedAt          time.Time     `gorm:"autoCreateTime" json:"created_at"`
	Payments           []Payment     `gorm:"foreignKey:LoanID" json:"payments,omitempty"`
}

// TableName 指定表名
func (Loan) TableName() string {
	return "loan"
}

File: frontend/src/types/loan.ts
export interface Loan {
  id: number;
  borrower_name: string;
  bank_name: string;
  total_amount: number;
  remaining_amount: number;
  interest_rate: number;
  start_date: string;
  end_date: string;
  first_payment_date?: string;
  payment_date: number;
  status: LoanStatus;
  payment_method: PaymentMethod;
  monthly_payment: number;
  created_at: string;
  payments?: Payment[];
}

export interface Payment {
  id: number;
  loan_id: number;
  amount: number;
  payment_date: string;
}

export type PaymentMethod = '等额本息' | '等额本金' | '先息后本';

export type LoanStatus = 'active' | 'completed' | 'overdue';

export const PaymentMethods = {
  EQUAL_INSTALLMENT: '等额本息' as PaymentMethod,
  EQUAL_PRINCIPAL: '等额本金' as PaymentMethod,
  INTEREST_FIRST: '先息后本' as PaymentMethod,
};

export const LoanStatuses = {
  ACTIVE: 'active' as LoanStatus,
  COMPLETED: 'completed' as LoanStatus,
  OVERDUE: 'overdue' as LoanStatus,
};
